### PR TITLE
Do not load old value for map set when map-store offload is enabled [HZ-1671] 

### PR DESCRIPTION
**Pull request by ahmetmircik**
related with #22601

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/map/impl/mapstore/MapStoreTest.java [hunk 1]
```diff
@@ -146,6 +146,34 @@ public Set<String> loadAllKeys() {
         assertFalse(loadCalled.get());
     }
 
+    @Test
+    public void map_set_does_not_load_old_value_when_offload_enabled() {
```
Comment (vbekiaris) [inline hazelcast/src/test/java/com/hazelcast/map/impl/mapstore/MapStoreTest.java:150]: I think it's better to test that "set does not load old value" here. Reasoning:
- "set does not load old value" is not specific to offload being enabled
-  force-offload builds will ensure the test is also executed with the offloaded code path.

Comment (ahmetmircik) [reply hazelcast/src/test/java/com/hazelcast/map/impl/mapstore/MapStoreTest.java:150]: Maybe we can remove the word offload from test name but anyway by default we always enable offloading for all map-store related tests.

Comment (vbekiaris) [reply hazelcast/src/test/java/com/hazelcast/map/impl/mapstore/MapStoreTest.java:150]: Yes, I would remove "offload" from the test name and not explicitly set offloading enabled in the mapstore config.
This way we can have this test executed with the defaults (offloading enabled) and when we put in place not-offloadable-by-default test job we can get it to test also the non-offloadable code paths.

**Review by frant-hartm — APPROVED** [review]

File: hazelcast/src/test/java/com/hazelcast/map/impl/mapstore/MapStoreTest.java (hunk 1 above)
Comment (ahmetmircik) [reply hazelcast/src/test/java/com/hazelcast/map/impl/mapstore/MapStoreTest.java]: done

**Review by vbekiaris — APPROVED** [review]

**Pull request merged**